sdk/tests/core/test_az_json.c

```diff
@@ -3105,17 +3105,6 @@ static void test_az_json_string_unescape(void** state)
     assert_true(az_span_is_content_equal(expected, destination_span));
   }
 
-  // insufficient space
-  {
-    az_span json = AZ_SPAN_FROM_STR(" { \"name\": \"some value string\" , \"code\" : 123456 } ");
```

Comment:
We need to be testing for `AZ_ERROR_NOT_ENOUGH_SPACE`, `AZ_ERROR_UNEXPECTED_CHAR`, `AZ_ERROR_UNEXPECTED_END`, for the null terminator at the end, for the precondition, for the correct size being written to `out_string_length`, for the precondition, and for the fact that the string is indeed unescaped where expected.

Maybe some of it is being tested, but it is not obvious so far. So far I can see you are deleting this coverage (one instance is legitimately a duplicate?) But I do not see at least the test for null terminator and for the precondition.

Reply:
The other cases you mentioned are covered already.

We haven't tested the other preconditions for this method, so we can improve coverage separately. The goal of the PR is just to fix the null terminator.

I will add the test case for nulls at the end.

Reply:
Here's the coverage. The null terminator line is covered:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=1769797&view=codecoverage-tab

And the `AZ_ERROR_NOT_ENOUGH_SPACE` case will only be possible if preconditions are off, which is why it isn't being covered now. We don't expect anyone to hit that case (and it is only there to avoid memory corruption, as an extra check).

I'll file an issue to address the test coverage for this API around preconditions, since the issue already exists in main.

Reply:
https://github.com/Azure/azure-sdk-for-c/issues/2310

Reply:
Is it even needed then, if it is a precondition? Once you violate the precondition, you're in the undefined behavior territory, and also the purposes of preconditions is for our code to not spend extra cycles checking for errors. I'd say it should be one or another.

Null terminator string will be shown as covered, but that's only because the line was executed during tests, and the tool can't see that you are not actually checking the buffer that null terminator was indeed being written. Please add that test.